### PR TITLE
app-arch/qpress: New package

### PR DESCRIPTION
**Pull request by winterheart**
This package required as runtime dependency for dev-db/percona-xtrabackup. 

When you have MySQL backup, compressed by xtrabackup, it complains that qpress command not found:

```
xtrabackup --decompress  --target-dir=base-comp
xtrabackup version 2.4.6 based on MySQL server 5.7.13 Linux (x86_64) (revision id: 8ec05b7)
170321 16:32:52 [01] decompressing ./mysql/host.MYD.qp
sh: qpress: command not found
cat: write error: Broken pipe
Error: thread 0 failed.
```

Package-Manager: Portage-2.3.3, Repoman-2.3.1

### CODE REVIEW

File: app-arch/qpress/qpress-1.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+src_prepare() {
+	# Fix compilation with newer gcc
+	sed -i '1i #include <unistd.h>' qpress.cpp || die
+	cp "${FILESDIR}/makefile" "${S}"
+
+	default
+}
+
+src_install() {
+	dobin qpress
```
Comment (monsieurp) [inline app-arch/qpress/qpress-1.1.ebuild]: qpress -> `${PN}`.

File: app-arch/qpress/qpress-1.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+RDEPEND="${DEPEND}"
+
+S="${WORKDIR}"
+
+src_prepare() {
+	# Fix compilation with newer gcc
+	sed -i '1i #include <unistd.h>' qpress.cpp || die
+	cp "${FILESDIR}/makefile" "${S}"
+
+	default
```
Comment (monsieurp) [inline app-arch/qpress/qpress-1.1.ebuild]: This needs to go right below `src_prepare()`.

File: app-arch/qpress/qpress-1.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+DEPEND=""
+RDEPEND="${DEPEND}"
+
+S="${WORKDIR}"
+
+src_prepare() {
+	# Fix compilation with newer gcc
+	sed -i '1i #include <unistd.h>' qpress.cpp || die
+	cp "${FILESDIR}/makefile" "${S}"
```
Comment (monsieurp) [inline app-arch/qpress/qpress-1.1.ebuild]: missing `|| die`.

File: app-arch/qpress/qpress-1.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+DESCRIPTION="A portable file archiver using QuickLZ algorithm"
+HOMEPAGE="http://www.quicklz.com/"
+SRC_URI="http://www.quicklz.com/${PN}-${PV/./}-source.zip"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+DEPEND=""
```
Comment (monsieurp) [inline app-arch/qpress/qpress-1.1.ebuild]: Nix this.

File: app-arch/qpress/qpress-1.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+RDEPEND="${DEPEND}"
```
Comment (monsieurp) [inline app-arch/qpress/qpress-1.1.ebuild]: This line too.

File: app-arch/qpress/qpress-1.1.ebuild [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+EAPI=6
+
+DESCRIPTION="A portable file archiver using QuickLZ algorithm"
+HOMEPAGE="http://www.quicklz.com/"
+SRC_URI="http://www.quicklz.com/${PN}-${PV/./}-source.zip"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
```
Comment (monsieurp) [inline app-arch/qpress/qpress-1.1.ebuild]: Please remove this line as well.

File: app-arch/qpress/qpress-1.1.ebuild [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+DEPEND=""
+RDEPEND="${DEPEND}"
+
+S="${WORKDIR}"
```
Comment (monsieurp) [inline app-arch/qpress/qpress-1.1.ebuild]: By default, `S` is equals to `${WORKDIR}`. No need to yet again define it.

Comment (winterheart) [inline app-arch/qpress/qpress-1.1.ebuild]: No, S is ${WORKDIR}/${P}. By default, compilation fails due 
```
The source directory '/var/tmp/portage/app-arch/qpress-1.1/work/qpress-1.1' doesn't exist
```

**Comment by winterheart** [comment]
@monsieurp fixed according you recommendations except ${S}.

**Comment by monsieurp** [comment]
Ok thanks. Will merge it ASAP.

**Pull request closed without merging**